compiler/src/typed/typedtree.re

```diff
@@ -21,6 +21,15 @@ open Types;
 let sexp_locs_disabled = _ => ! Grain_utils.Config.sexp_locs_enabled^;
 
 type loc('a) = Location.loc('a);
+
+[@deriving sexp]
+type attributes = list(attribute)
+
+[@deriving sexp]
+and attribute =
+  | Disable_gc
+  | External_name(string);
```

Comment:
Why are these in typedtree now? Just curious

Reply:
It felt really bad having to search through strings and assume the arguments rather than look at nice enums. I moved it here and got rid of a ton of `failwith("impossible")`s 🤘

Reply:
Love it. I was just wondering why you chose the typedtree specifically? Idk how we usually determine where things should be declared.

Reply:
Ah. I think of the parsetree as the wild west, where anything goes but nothing is confirmed (you can write `void + true` if you want, we don't care etc., the formatter will even format it for you), but the typedtree is where things are settled—we know both sides of `+` are numbers, functions are called with the right number of arguments, etc. In a similar vein, this is sort of the typechecked version of attributes. This is the first point in time that we _know_ that the attributes will have the right number of args, etc, so we might as well encode that in here rather than waiting until later in the stages of the compiler/accidentally messing it up.

That's my thought process, at least.